src/mbgl/renderer/layers/render_symbol_layer.cpp

```diff
@@ -580,7 +582,32 @@ void RenderSymbolLayer::prepare(const LayerPrepareParameters& params) {
             const Tile* tile = params.source->getRenderedTile(renderTile.id);
             assert(tile);
             assert(tile->kind == Tile::Kind::Geometry);
-            placementData.push_back({*bucket, renderTile, static_cast<const GeometryTile*>(tile)->getFeatureIndex()});
+
+            bool firstInBucket = true;
+
+            if (bucket->sortKeyRanges.size() == 0) {
+                placementData.push_back({*bucket,
+                                         renderTile,
+                                         static_cast<const GeometryTile*>(tile)->getFeatureIndex(),
+                                         firstInBucket,
+                                         0.0f,
+                                         0,
+                                         bucket->symbolInstances.size()});
+            } else {
+                for (const SortKeyRange& sortKeyRange : bucket->sortKeyRanges) {
+                    LayerPlacementData layerData{*bucket,
+                                                 renderTile,
+                                                 static_cast<const GeometryTile*>(tile)->getFeatureIndex(),
```

Comment:
nit: might make sense to cache `static_cast<const GeometryTile*>(tile)->getFeatureIndex()` in a local variable